### DISCUSSION TITLE
Understanding various options in Settings

### DISCUSSION THREAD

**Discussion by Macaroni108** [apohl79/audiogridder#614] [General]
I haven't been able to find details about several options in Settings for AudioGridder Server.

Local Mode - what exactly does this do re plugins within the AG plugin?

Allow plugins to be loaded in parallel - is that within the AG plugin, and if so, how do we know if it's series or parallel, ie: where do we choose which one?

I see there is an AudioGridderPluginTray app. How is that used? Is it in conjunction with Sandboxing in Settings? If so, how should that be set up - or is it automatic?

**Comment by apohl79** [comment]
> I haven't been able to find details about several options in Settings for AudioGridder Server.

The documentation needs updates regarding this.

> Local Mode - what exactly does this do re plugins within the AG plugin?

Local mode does nothing to the plugins. It is just to improve working with AG on the same machine. The plugin windows will be attached to the AG DAW plugin window. When you use it with sandboxing, you will also be able to have multiple plugin UIs simultaneously.

> Allow plugins to be loaded in parallel - is that within the AG plugin, and if so, how do we know if it's series or parallel, ie: where do we choose which one?

Ignore this. When you have sandboxing disabled, all plugins run within the same process in separate threads. I added this option to experiment with the stability, as some plugins are crashing the server sometimes when loading a session, where many plugins get loaded in parallel. In the end it does not make things better or worse, so leave it on. I will remove this option.

> I see there is an AudioGridderPluginTray app. How is that used? Is it in conjunction with Sandboxing in Settings? If so, how should that be set up - or is it automatic?

Once you load an AG plugin into your DAW, it will start the tray app automatically. The tray app on the other side will terminate automatically when there is no plugin connected.

**Comment by Macaroni108** [comment]
> Local mode does nothing to the plugins. It is just to improve working with AG on the same machine. The plugin windows will be attached to the AG DAW plugin window. When you use it with sandboxing, you will also be able to have multiple plugin UIs simultaneously.
> 

I tried the Sandboxing option once and saw that there were multiple AudioGridder servers in Activity Monitor on my iMac. It wasn't as stable as without it, but I may not have had the other settings correct. Is Sandboxing the most efficient and preferred method to use?

> Allow plugins to be loaded in parallel - is that within the AG plugin, and if so, how do we know if it's series or parallel, ie: where do we choose which one?
 
 ...so leave it on. I will remove this option.
>

It was off by default. Should I turn it on until you remove it?

Thanks so much for your feedback. I appreciate it. 

This has fantastic potential once it's stabilized and optimized. Is there a new beta coming soon?

**Reply by apohl79** [reply]
> I tried the Sandboxing option once and saw that there were multiple AudioGridder servers in Activity Monitor on my iMac. It wasn't as stable as without it, but I may not have had the other settings correct. Is Sandboxing the most efficient and preferred method to use?

I suggest you try if it helps with your setup. For me, I see much better session load times and stability, as each AG plugin connects to a dedicated sandbox process. If a plugin crashes, it affects only the single sandbox. But that also means, that plugins, that lookout for other instances in the same process space for interaction will just run in isolation.

> It was off by default. Should I turn it on until you remove it?

Yes you can. It does not make much of a difference.

> Is there a new beta coming soon?

I have no ETA right now. Once I can focus more on AG again, I will address the reported issues and provide a new beta.

**Reply by mixmonkey** [reply]
Re: Sandboxing option, I get frequent crashes when removing plugins when I have Sandboxing on, so left it off.
